### ISSUE TITLE
Issues pulling Intraday historical data

### ISSUE THREAD

**Issue by drbh** [alpacahq/alpaca-trade-api-go#120]
Hey Alpaci,

I'm trying to pull intraday historical data via `GetBars` and I'm getting inconsistent results...

TLDR; my goal is to create a script that allows me to fetch all of the 1 minute intraday data as far back as possible

#### Doesn't return data unless the date range is large enough?

If I try to fetch data between `2020-12-25 13:00:00` and `2020-12-30 13:00:00` then I get no results... but If I fetch data between `2020-12-24 13:00:00` and `2020-12-30 13:00:00` I get `1990` items.

Is there is a reason that the first one would fail and the second one would success right after?

#### Missing minutes sometimes hours

Most of the calls succeed and data is returned - however upon some manual inspection I noticed some missing minutes (I assumed this was minutes without any transactions) but theres many minutes missing in a row and it doesnt seem likely that there was no transaction for up to 55 mins

| Open | High | Low | Close | Vol | Time |
| --- | --- | --- | --- | --- | --- |
| 219.3 | 219.3 | 219.2 | 219.2 | 129 | 2021-03-04T04:05:00-05:00 |  
| 219.31 | 219.31 | 219.31 | 219.31 | 100 | 2021-03-04T04:02:00-05:00 |  
| 219.2 | 219.2 | 219.2 | 219.2 | 20 | 2021-03-04T04:01:00-05:00 |  
| 219.05 | 219.11 | 219.05 | 219.11 | 450 | 2021-03-04T04:00:00-05:00 |  
| 219.3 | 219.3 | 219.3 | 219.3 | 420 | 2021-03-04T03:05:00-05:00 |  
| 219.14 | 219.14 | 219.14 | 219.14 | 2 | 2021-03-04T03:04:00-05:00 |  
| 219.11 | 219.11 | 219.11 | 219.11 | 100 | 2021-03-04T02:45:00-05:00 |  

Notice the gap between `2021-03-04T04:00:00-05:00` and `2021-03-04T03:05:00-05:00`. That's :55 minutes between 3:05 and 4:00....

Is there any reason large chunks of data are missing?

#### The script in question

```golang
package main

import (
	"fmt"
	"github.com/davecgh/go-spew/spew"
	"github.com/fatih/structs"
	"os"
	"reflect"
	"sort"
	"time"

	"github.com/alpacahq/alpaca-trade-api-go/alpaca"
	"github.com/alpacahq/alpaca-trade-api-go/common"
	v2 "github.com/alpacahq/alpaca-trade-api-go/v2"
)

func init() {
	os.Setenv(common.EnvApiKeyID, "XXXX")
	os.Setenv(common.EnvApiSecretKey, "XXXX")
	fmt.Printf("Running w/ credentials [%v %v]\n", common.Credentials().ID, common.Credentials().Secret)
}

func main() {
	alpacaClient := alpaca.NewClient(common.Credentials())

	end := time.Now().AddDate(0, 0, -1) // yesterday
	start := end.AddDate(0, 0, -5)      // Subtract N Day

	// holder for all data
	mybars := []v2.Bar{}
	counter := 0
	for {

		if counter > 100 {
			break
		}

		bars := alpacaClient.GetBars(
			"IWM",
			v2.Min,
			v2.Raw,
			start,
			end,
			10000,
		)

		innerBars := []v2.Bar{}
		// iterate over channel
		for msg := range bars {
			innerBars = append(innerBars, msg.Bar)
		}

		// first and last
		startBar := innerBars[0]
		fmt.Printf("(%d)\t\t(%d)\t(%d)\t(%.2f)\t%s - %s\n", len(mybars), counter, len(innerBars), start.Sub(end).Hours()/24, start, end)
		if len(innerBars) < 2 {
			// increment fail counter
			counter += 1
			// push start day back one day
			start = start.AddDate(0, 0, -1)
		} else {
			end = startBar.Timestamp
			start = end.AddDate(0, 0, -5) // Subtract N Day
			mybars = append(mybars, innerBars...)

		}
		// short sleep
		time.Sleep(time.Millisecond * 100)
	}

	// sort them so the dates are in order
	sort.SliceStable(mybars, func(i, j int) bool {
		return mybars[i].Timestamp.After(mybars[j].Timestamp)
	})

	// write all data to file
	WriteBars(mybars, "intra.csv")
}

func WriteBars(bars []v2.Bar, name string) {

	//init the loc
	loc, _ := time.LoadLocation("America/New_York")

	file, fileErr := os.Create(name)
	if fileErr != nil {
		spew.Dump(fileErr)
	}
	names := structs.Names(bars[0])
	for i := 0; i < len(names); i++ {
		// values[i] = v.Field(i).Interface()
		fmt.Fprintf(file, "%s", names[i])
		if i < (len(names) - 1) {
			fmt.Fprintf(file, ",")
		}
	}
	fmt.Fprintf(file, "\n")

	for i := range bars {
		k := bars[i]
		v := reflect.ValueOf(k)

		// write each value type accordingly
		for i := 0; i < v.NumField(); i++ {
			if i < 4 {
				// these are all floats
				fmt.Fprintf(file, "%.2f", v.Field(i).Interface())
			} else if i == 4 {
				// this is a usize
				j := int(v.Field(i).Interface().(uint64))
				fmt.Fprintf(file, "%d", j)
			} else {
				// this is a UTC time.Time
				t := v.Field(i).Interface().(time.Time)
				fmt.Fprintf(file, "%s", t.In(loc).Format(time.RFC3339))
			}

			// if its not the last column add a comma
			if i < (v.NumField() - 1) {
				fmt.Fprintf(file, ",")
			}
		}
		fmt.Fprintf(file, "\n")
	}
}
```

UPDATE: 

Heres a link to the csv file the above produced: https://docs.google.com/spreadsheets/d/1nVsGPuTGg_XTQZDx8aBawNBsgAvroijWM55hsfAo7Ug/edit?usp=sharing

**Comment by gnvk** [comment]
Please check this now using v2. If you still find an issue, reopen the ticket. Thanks!
